Yoast/wordpress-seo
# Wizard notice to notification

## Context


* Remove the wizard notice on the dashboard and refactor it into a notification in the notification centre.

## Summary


This PR can be summarized in the following changelog entry:

* Creates a new class `class-wizard-notification` which can be accessed to get the appropriate notification depending on the state. Also removes the wizard notice completely

## Relevant technical choices:

**IMPORTANT NOTICE
There is no set date when the admin-redesign branch is merged. Therefore, the version for this upgrade routine in `class-upgrade.php` is set to 14.4 because it will probably be merged around that time, however this may change in the future.**

* The notification is built up with 3 states:
  0 - The wizard has not been started and the notification is a call to action to open it
  1 - The wizard has been opened but has not been finished, the notification is a call to action to 
  open it
  2 - The wizard has been finished and the notification hints that you can reopen it to change 
  some settings with a call to action
The class `class-wizard-notification` takes care of these notifications
* The old function notification in `class-configuration-notifier.php` has been removed which styled the wizard notice before but is not applicable in this approach

## Test instructions

This PR can be tested by following these steps:

Testing if the old unused db meta is removed:
**QA can skip the first step, please double check if this version is correct (14.4-RC0) **
1. In class-upgrade.php, change line 56 to `'14.4-RC0'   => 'upgrade_144',` (line 56 is the )
2. Checkout this branch, build and activate the plugin
3. Install and configure the yoast test helper
4. Open the yoast test helper and make sure the db version is set to `14.0-RC4`
5. Open a DB browser (i used sequel pro), and in the table `wp_usermeta` search for a field with `meta_key` to have the value of `wpseo-dismiss-configuration-notice`
6. Make sure it exists
7. Go to the yoast test helper again, and set the db version to `13.0-RC0` and hit save
8. Open the db browser again and make sure the field searched for earlier is gone

Testing the functionality:
Step 1 and 2 below are already done in the above test instructions, but are mentioned just in case.
1. Checkout this branch, build and activate the plugin
2. Install and configure the yoast test helper
  **Update by Andrea:** the test helper plugin needs to be switched to the `configuration-wizard-update` branch.
3. Go to Tools-> Yoast Test and click the "Reset Configuration Wizard" button
4. Go to the Yoast SEO plugin dashboard
5. Check if you have the following notification:
![Screenshot 2020-04-28 at 10 45 24](https://user-images.githubusercontent.com/9864796/80466868-60e74100-893d-11ea-96f4-2e58992a3aa7.png)
6. Open the configuration wizard and click the `CLOSE THE WIZARD` button
7. Check if the above notification is gone and you now have this notification:
![Screenshot 2020-04-24 at 12 56 50](https://user-images.githubusercontent.com/9864796/80205538-127b2f00-862b-11ea-92b8-17342becba3a.png)
8. Open the wizard again, go to the last page and click the `Close` button
9. You are redirected to the dashboard, refresh that page
10. Check if the above notifications are gone and you now have the following notification:
![Screenshot 2020-04-24 at 12 58 55](https://user-images.githubusercontent.com/9864796/80205712-5d954200-862b-11ea-9837-f0abbc223f3b.png)


## UI changes

## Documentation

## Quality assurance


Fixes #14800
